jaxrs/src/main/java/com/cerner/beadledom/jaxrs/exceptionmapping/WebApplicationExceptionMapper.java

```diff
@@ -0,0 +1,77 @@
+package com.cerner.beadledom.jaxrs.exceptionmapping;
+
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+
+import com.cerner.beadledom.json.common.model.JsonError;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An {@link ExceptionMapper} for the {@link WebApplicationException} family of exceptions.
+ *
+ * <p>The intention of this exception mapper is to restructure the exceptions into a standard JSON
+ * format.
+ *
+ * @author Cal Fisher
+ * @since 2.6
+ */
+@Provider
+public class WebApplicationExceptionMapper implements ExceptionMapper<WebApplicationException> {
+
+  private static final Logger logger = LoggerFactory.getLogger(WebApplicationExceptionMapper.class);
+
+  /**
+   * Maps an unhandled {@link WebApplicationException} to a {@link Response}.
+   *
+   * @param exception the {@link WebApplicationException} that was not handled.
+   * @return a {@link Response} object with a status of the {@link WebApplicationException} or 500
+   *     if the exception's response is null, content-type of 'application/json', and a
+   *     {@link JsonError} entity containing details about the unhandled exception in JSON format.
+   */
+  @Override
+  public Response toResponse(WebApplicationException exception) {
+
+    Response response = exception.getResponse();
+    int code = response == null ? INTERNAL_SERVER_ERROR.getStatusCode() : response.getStatus();
+    Response.Status status = Response.Status.fromStatusCode(code);
+
+    if (status == null) {
```

Comment:
After some offline conversation I don't think this is the behavior we want. `Response.Status` only supports the core http status codes. It doesn't have representation for any of the extensions. As an example, with the code as it is we would end up mapping `422` to a `500`.

Our real issue here, is when `Response.Status#fromStatusCode` returns `null` we won't have a status phrase to use for the message. We could use `Response.Status#getFamily` and have some message we use depending of the family of the status?

@johnlcox @shiftyeyes

Reply:
That sounds good. Use the standard message if one is found; if not fall back to the family and use the family to put a more generic message like `Client Error` and `Server Error`.

Reply:
Added a method called `getMessage(int code)` that will return the message using `Response.Status#fromStatusCode` unless null is returned, then determines the message using `Response.Status#getFamily`: https://github.com/cerner/beadledom/pull/45/commits/cf92faf55674bfa402abe51eff5459464d3ba663#diff-a3d460ba2485f948686ff5823fe46af8R67